src/Client.js

```diff
@@ -275,27 +275,26 @@ export default class Client {
 
     // sendCommand :: Command -> Number -> Promise Command
     sendCommand(command, timeout = this._application.commandTimeout) {
-        var commandPromise = Promise.race([
+        const commandPromise = Promise.race([
             new Promise((resolve, reject) => {
                 this._commandResolves[command.id] = (c) => {
```

Comment:
To prevent a memory leak caused by an error in this function execution, you can design it in that way:

```
() => {
    try {
        // logic here
    } catch (err) {
        return reject(...);
    } finally {
        delete this._commandResolves[command.id];
    }
}
```

Reply:
We actually need a race of 2 promises here, because the command must fail if there was no response before the timeout

Reply:
I didn't question that. I'm talking about the arrow function you pass on the first promise.

Reply:
The try/catch block is a good way to prevent unhandled errors. By the way, it might help us to handle exceptions better at BLiP Portal